lib/web/apiserver.go

```diff
@@ -361,14 +361,38 @@ func (m *Handler) getAuthenticationSettings(w http.ResponseWriter, r *http.Reque
 	return pr, nil
 }
 
+type webConfig struct {
+	Auth interface{} `json:"auth"`
+	// ServerVersion is the version of Teleport that is running.
+	ServerVersion string `json:"serverVersion"`
+}
+
 // getConfigurationSettings returns configuration for the web application.
 func (m *Handler) getConfigurationSettings(w http.ResponseWriter, r *http.Request) (interface{}, error) {
-	pr, err := buildAuthenticationSettings(m.cfg.ProxyClient)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	webCfg := webConfig{ServerVersion: teleport.Version}
+	as := client.AuthenticationSettings{}
+	cap, err := m.cfg.ProxyClient.GetClusterAuthPreference()
+
+	if err == nil {
```

Comment:
What do you think about the following instead:

```go
func buildAuthenticationSettings(authClient auth.ClientI) (*client.AuthenticationSettings, error) {
	as := client.AuthenticationSettings{}

	cap, err := authClient.GetClusterAuthPreference()
	if err != nil {
		return nil, trace.Wrap(err)
	}

	as.Type = cap.GetType()
	as.SecondFactor = cap.GetSecondFactor()

	// if we have u2f or oidc settings, build those as well
	if cap.GetSecondFactor() == teleport.U2F {
		as.U2F = buildUniversalSecondFactorSettings(authClient)
	}
	if cap.GetType() == teleport.OIDC {
		as.OIDC = buildOIDCConnectorSettings(authClient)
	}

	return as, nil
}

func (m *Handler) getAuthenticationSettings(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
 	as, err := buildAuthenticationSettings(m.cfg.ProxyClient)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
	return &client.PingResponse{
 		Auth:          as,
 		ServerVersion: teleport.Version,
 	}, nil
  }

func (m *Handler) getConfigurationSettings(w http.ResponseWriter, r *http.Request) (interface{}, error) {
   	[...]
 
	as, err = buildAuthenticationSettings(m.cfg.ProxyClient)
	if err != nil {
		log.Infof("Cannot retrieve cluster auth preferences: %v", err)	
	}
	webCfg.Auth = as

 	[...]
```

This way we don't duplicate the code to build authentication settings.